Detectors/MUON/MCH/Simulation/src/Digitizer.cxx

```diff
@@ -133,54 +147,38 @@ int Digitizer::processHit(const Hit& hit, double event_time)
 
   //need to keep both electros separated since afterwards signal generation on each plane
   //how misalignment enters?
-  std::vector<int> padIDsbend;
-  std::vector<int> padIDsnon;
+  std::vector<int> padIDs;
 
   //retrieve pads with signal
-  mSeg[indexID].bending().forEachPadInArea(xMin, yMin, xMax, yMax, [&padIDsbend](int padid) { padIDsbend.emplace_back(padid); });
-  mSeg[indexID].nonBending().forEachPadInArea(xMin, yMin, xMax, yMax, [&padIDsnon](int padid) { padIDsnon.emplace_back(padid); });
+
+  mSeg[indexID].forEachPadInArea(xMin, yMin, xMax, yMax, [&padIDs](int padid) { padIDs.emplace_back(padid); });
```

Comment:
why are you filling an array of padids and looping over it, instead of doing : 

mSeg.forEachPadInArea() .... { xmin = ... } 

simply ?

Reply:
Hi,
I am not experienced with this lambda functions, hence my ignorance.
However, when I was trying to put it now in a single looping over the pads in each area, I get problems with all the member functions that I need to call inside that loop. 
I.e., I would need to do something like: 

 mSeg[indexID].forEachPadInArea(xMin, yMin, xMax, yMax, [&anodpos, &xmin, &xmax, &ymin, &ymax, &chargebend, &chargenon, &signal, &ndigits, &mDigits, response, labelIndex](int padid) 
(still missing the mSeg[index])
so I am not sure, but I far as I understand, I have to copy the response in the same scope of the lambda,, I would need to do a copy of the objects.  For the moment, you could maybe tell me if I misunderstood you and whether I there are easier ways. Otherwise, if I don't put everything inside this forEachPadInArea function (where I have to copy this response function for), I have to loop anyhow once more.